torch/distributed/_composable/fsdp/_fsdp_init.py

```diff
@@ -83,3 +84,31 @@ def _get_managed_states(
             buffers.append(buffer)
             visited_buffers.add(buffer)
     return params, buffers
+
+
+def _move_states_to_device(
+    params: List[nn.Parameter],
+    buffers: List[torch.Tensor],
+    device: torch.device,
+    mesh_info: FSDPMeshInfo,
+) -> None:
+    """
+    We have FSDP move states to device for simpler and faster initialization
+    since FSDP almost always uses CUDA for training. We move parameters/buffers
+    rather than modules to allow ignoring specific states in the future.
```

Comment:
Nit: I still spend some time to understand the differernce between ``module.to(device)`` and ``param/buffer.to(device)``. 

maybe a even more straightforward docstring

```We move parameters/buffers rather than modules since modules may contain ignored parameters/buffers```